### PR TITLE
[Fixes #1638, #1639] UI fixes + tests for social login

### LINKED ISSUE #1638: Send confirmation email doesn't generate the callback URI correctly
* On the manage/index page. The send email link sends an email confirmation message with an incorrect callback URI of id=<<user-id>> instead of userId=<<userId>> causing email confirmation to fail.
* The email confirmation handlers redirects the user using RedirectToPage("/Index") which is not correct for MVC based pages.

### PR DESCRIPTION
**Pull request by javiercn**
* Added a "fake social login provider"
* Refactored test infrastructure.
* Added tests for social login.
* Fixed bug in email confirmation through the management UI.
* Fixed bug in password reset.

### CODE REVIEW

File: src/UI/Areas/Identity/Pages/Account/ResetPassword.cshtml.cs [hunk 1]
```diff
@@ -31,12 +31,13 @@ public class InputModel
...
             public string ConfirmPassword { get; set; }
 
+            [Required]
             public string Code { get; set; }
 
-            public virtual IActionResult OnGet(string code = null) => throw new NotImplementedException();
-
-            public virtual Task<IActionResult> OnPostAsync() => throw new NotImplementedException();
         }
+        public virtual IActionResult OnGet(string code = null) => throw new NotImplementedException();
```
Comment (HaoK) [inline src/UI/Areas/Identity/Pages/Account/ResetPassword.cshtml.cs]: Is there any way to detect this defect via some kind of reflection test?  Is it safe to say that every page in the default UI should have these two as public virtuals?

File: src/UI/Areas/Identity/Pages/Account/Manage/Index.cshtml.cs [hunk 2]
```diff
@@ -132,7 +132,7 @@ internal class IndexModel<TUser> : IndexModel where TUser : IdentityUser
             var callbackUrl = Url.Page(
                 "/Account/ConfirmEmail",
                 pageHandler: null,
-                values: new { user.Id, code },
+                values: new { userId = user.Id, code = code },
```
Comment (HaoK) [inline src/UI/Areas/Identity/Pages/Account/Manage/Index.cshtml.cs]: Was this a root cause of one of the bugs?

File: src/UI/Areas/Identity/Pages/Account/ResetPassword.cshtml.cs (hunk 1 above)
Comment (javiercn) [reply src/UI/Areas/Identity/Pages/Account/ResetPassword.cshtml.cs]: I'm trying to find a way of detecting this at run time, but I haven't come up with something that is not "flaky". I'll chat with Ryan about it.

File: src/UI/Areas/Identity/Pages/Account/Manage/Index.cshtml.cs (hunk 2 above)
Comment (javiercn) [reply src/UI/Areas/Identity/Pages/Account/Manage/Index.cshtml.cs]: Yes, the parameter being sent was not correct.

**Pull request merged**